doc/reduce.md

```diff
@@ -12,30 +12,104 @@ grppi::reduce(exec, other_arguments...);
 
 There is a single variant of the reduction:
 
-* Sequence reduction: Takes a sequence represented by two iterators and reduces the values.
+* **Sequence reduction without initial value**: Reduces a sequence of values
+with no initial value.
+
+* **Sequence reduction with initial value**: Reduces a sequence of values with
+an initial value.
 
 
 ## Key elements in a reduction
 
-The key element of a reduction is the combination **Operation**. The operation may be any C++ callable entity. This operation, is a binary operation taking two values of the same type and combining them into a value of the same type.
+The key element of a reduction is the **Combiner** operation. The **Combiner**
+may be an **HeterogeneousCombiner** or an **HomogeneousCombiner**.
+
+A **Combiner** is any C++ callable entity, that is able to combine two values
+into a single value. The requirements for the **Combiner** are different
+depending on whether the reduction uses an initial value
+(**HeterogeneousCombiner**) or not (**HomogeneousCombiner**).
+
+An **HomogeneousCombiner** `cmb` is any operatingo taking two values `x`and `y`
+of type `T` and returning a combined value of type `T`, making valid the
+following.
+
+~~~{.cpp}
+T x, y;
+T res = cmb(x,y);
+~~~
+
+An **HeterogeneousCombiner** `cmb` is any operations taking two values `x` and
+`y` of types `T` and `U` and returning a combined value of type `T`, making valid
+the following:
+
+~~~{.cpp}
+T x;
+U y;
+T res = cmb(x,y);
+~~~
 
 ## Details on reduction variants
 
-### Sequence reduction
+### Sequence reducction without identity
```

Comment:
This subsection should be removed as this interface is not supported in the current version.